### ISSUE TITLE
How to extract full article instead of just abstract?

### ISSUE THREAD

**Issue by monk1337** [metapub/metapub#31]
Hi, Thank you for this awesome tool. I am trying to extract full article instead of just abstract but couldn't find any method in metapub classes?

These ara all methods supported by article module :

`dict_keys(['pubmed_type', '_root', 'xml', 'content', 'pmid', 'url', 'authors', 'author_list', 'title', 'authors_str', 'author1_last_fm', 'author1_lastfm', 'pages', 'first_page', 'last_page', 'volume', 'issue', 'volume_issue', 'doi', 'pii', 'pmc', 'issn', 'mesh', 'chemicals', 'grants', 'publication_types', 'book_accession_id', 'book_title', 'book_publisher', 'book_language', 'book_editors', 'book_abstracts', 'book_sections', 'book_copyright', 'book_medium', 'book_synonyms', 'book_publication_status', 'book_history', 'book_contribution_date', 'book_date_revised', 'abstract', 'journal', 'year', 'history'])`

How I can extract full article text (especially introduction )?

**Comment by nthmost** [comment]
Hi @monk1337 ,

Unfortunately this is a complicated issue.  The NCBI doesn't make it natively easy to grab the full text due to journal licensing agreements and such.  Whatever fulltext is available via PMC is only supposed to be available programmatically via the Open Access Subset.

You can read about this situation here:  https://www.ncbi.nlm.nih.gov/pmc/tools/get-full-text/

What this means is that eutils (the underlying API that metapub uses) cannot accomplish this.

However...

This is why I made FindIt.  😉 

If your article is available in fulltext, you will be able to do the following:

```
from metapub import FindIt

pmid = "your_pmid_here"
src = FindIt(pmid)
if src.url:
    # insert your favorite method of downloading a file here, for example:
    # https://stackoverflow.com/questions/7243750/download-file-from-web-in-python-3
```

Basically, I can't in good conscience build a fulltext-downloading function into the PubMedArticle object, as it is meant to be a faithful API wrapper for eutils.

However, metapub definitely includes all the tools you might need for text mining -- even articles outside of the PMC are grabbable with FindIt.

Let me know if you'd like more advice!
